tests/CMakeLists.txt

```diff
@@ -76,3 +82,20 @@ endif()
 
 include_directories(${CMAKE_SOURCE_DIR}/src ${PROJECT_BINARY_DIR})
 configure_file("${PROJECT_SOURCE_DIR}/tests/config.h.in" "${PROJECT_BINARY_DIR}/tests/config.h" ESCAPE_QUOTES @ONLY)
+
+if (LIBPAM_HAVE_CONFDIR)
+
+    #compile PAM test module
+    set(CMAKE_SHARED_LIBRARY_PREFIX  )
+    add_library(pam_netconf SHARED ${CMAKE_SOURCE_DIR}/tests/pam/pam_netconf.c)
+    target_link_libraries(pam_netconf ${LIBPAM_LIBRARIES})
+
+    #generate PAM configuration file
+    FILE(WRITE ${CMAKE_SOURCE_DIR}/tests/pam/netconf.conf
+    "#%PAM-1.4\n"
+    "auth required ${CMAKE_SOURCE_DIR}/build/tests/pam_netconf.so\n"
+    "account required ${CMAKE_SOURCE_DIR}/build/tests/pam_netconf.so\n"
+    "password required ${CMAKE_SOURCE_DIR}/build/tests/pam_netconf.so\n"
+    )
```

Comment:
When writing a command on several lines, always increase the indent on the other lines. Also, `cmake` is case insensitive and we use the lowercase command names, so use just `file`.